pkg/protocol/engine/accounts/accountsledger/manager.go

```diff
@@ -355,6 +355,33 @@ func (m *Manager) AddAccount(output *utxoledger.Output, blockIssuanceCredits iot
 	return nil
 }
 
+// Reset resets the component to a clean state as if it was created at the last commitment.
+func (m *Manager) Reset() {
+	blockBurnsToDelete := make([]iotago.SlotIndex, 0)
+	m.blockBurns.ForEachKey(func(slot iotago.SlotIndex) bool {
+		if slot > m.latestCommittedSlot {
+			blockBurnsToDelete = append(blockBurnsToDelete, slot)
+		}
+
+		return true
+	})
+
+	versionSignalsToDelete := make([]iotago.SlotIndex, 0)
+	m.latestSupportedVersionSignals.ForEach(func(slot iotago.SlotIndex, _ *shrinkingmap.ShrinkingMap[iotago.AccountID, *model.SignaledBlock]) {
+		if slot > m.latestCommittedSlot {
+			versionSignalsToDelete = append(versionSignalsToDelete, slot)
+		}
+	})
+
+	for _, slot := range blockBurnsToDelete {
+		m.blockBurns.Delete(slot)
```

Comment:
Why not deleting them straight away instead of collecting them first?

Reply:
Because the foreach grabs a read lock which prevents us from writing to the collection in the same step.